src/mlpack/methods/ann/ffn.hpp

```diff
@@ -394,11 +394,13 @@ class FFN
 
   //! Locally-stored copy visitor
   CopyVisitor<CustomLayers...> copyVisitor;
+
+template<typename Model, typename InitializerType, class NoiseType>
```

Comment:
Another picky style comment, can we tab this ones? And do you think we should add something like ``The GAN class should have access to internal members.``?